### PR TITLE
Fix Multiple Matching Tokens Bug

### LINKED ISSUE #2402: Calling two backend APIs not possible with interceptor => "The cache contains multiple tokens satisfying the requirements"
## Library
- [x ] `msal@1.4.1`
- [ x] `@azure/msal-angular@1.1.1`

## Framework
Angular 10.1.4

## Description
If I try to call two different APIs from my Angular application, msal throws an error that the token cache contains multiple tokens satisfying the requirements.
In both cases I have the same client id and authority. Only the scopes are different as you can see in my configuration.
If I delete the token cache and reload my page, it works for a single time. After reloading the page the error message appears again.
I am using the http interceptor to protect my API calls. 

## Error Message
ClientAuthError: Cache error for scope openid,profile: The cache contains multiple tokens satisfying the requirements. Call AcquireToken again providing more requirements like authority..
ClientAuthError: Cache error for scope openid,profile: The cache contains multiple tokens satisfying the requirements. Call AcquireToken again providing more requirements like authority..

## MSAL Configuration
```
MsalModule.forRoot({
      auth: {
        clientId: 'd51ec65d-21a3-4859-ac1c-dd833e524d85',
        authority: 'https://login.microsoftonline.com/organizations/',
        redirectUri: 'http://localhost:4200',     
        validateAuthority: true,
        postLogoutRedirectUri: "http://localhost:4200/",
        navigateToLoginRequestUrl: true,           
      },
      cache: {
        cacheLocation: 'localStorage',
        storeAuthStateInCookie: isIE, // set to true for IE 11
      },
    },
    {
      popUp: isIE,
      consentScopes: [
        'user.read',
        'openid',
        'profile',
        'api://d51ec65d-21a3-4859-ac1c-dd833e524d85/areas.read.all'
      ],
      unprotectedResources: [],
      protectedResourceMap: [
        ['https://graph.microsoft.com/v1.0', ['user.read', 'user.read.all', 'openid']],
        ['https://localhost:5001', ['api://d51ec65d-21a3-4859-ac1c-dd833e524d85/areas.read.all']],
      ],
      extraQueryParameters: {}
    })
```js
// Provide configuration values here.
// For Azure B2C issues, please include your policies.
```

## Reproduction steps
Try to call graph API and your own API within one angular project with msal configured.

## Expected behavior
The token for my own API is used for calls to localhast... the graph token for calling the graph. This should work OOTB.
## Browsers/Environment
- [x ] Chrome

### PR DESCRIPTION
**Pull request by tnorling**
- Addresses a bug causing calls to `acquireTokenSilent` for an id_token to throw a "multiple_matching_tokens" error
- Also replaces this error with a log message, remove matching tokens from cache and return null so that in the future if something causes this to break again it doesn't break the application flow but rather goes to the network to get a new token instead.

Closes #2402

### CODE REVIEW

File: change/msal-2020-11-10-11-43-41-fix-multiple-matching-tokens.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "type": "patch",
+  "comment": "Fix multiple matching tokens error (#2582)",
+  "packageName": "msal",
+  "email": "thomas.norling@microsoft.com",
+  "dependentChangeType": "patch",
```
Comment (tnorling) [inline change/msal-2020-11-10-11-43-41-fix-multiple-matching-tokens.json:6]:
```suggestion
  "dependentChangeType": "none",
```

**Review by hectormmg — APPROVED** [review]
Looks good to me! Thanks for taking this!

**Review by jasonnutter — APPROVED** [review]

**Comment by bdbvb** [comment]
ETA on 1.4.4 milestone?

**Comment by tnorling** [comment]
@bdbvb version 1.4.4 is now available

**Pull request merged**